src/main/java/org/jenkinsci/maven/plugins/hpi/RunMojo.java

```diff
@@ -612,19 +651,13 @@ public void addJars(org.eclipse.jetty.util.resource.Resource lib) {
         }
     }
 
-    public JettyServer createServer() throws Exception {
-        return new JettyServer() {
-            @Override
-            public Connector createDefaultConnector(String portnum) throws Exception {
-                if ((portnum == null || portnum.equals("")) && defaultPort != null) {
-                    portnum = defaultPort;
-                }
-                SelectChannelConnector con = (SelectChannelConnector)super.createDefaultConnector(portnum);
-                con.setResponseHeaderSize(12*1024);// use a bigger buffer as Stapler traces can get pretty large on deeply nested URL
-
-                return con;
-            }
-        };
+    @Override
+    public void startJetty() throws MojoExecutionException {
+        if (httpConnector==null && defaultPort!=0) {
+            httpConnector = new MavenServerConnector();
+            httpConnector.setPort(defaultPort);
+        }
```

Comment:
Maybe you meant this to be something like...

if (httpConnector==null) {
    httpConnector = new MavenServerConnector();
}
if (defaultPort!=0) {
    httpConnector.setPort(defaultPort);
}